akka-typed-tests/src/test/scala/akka/typed/cluster/sharding/ClusterShardingSpec.scala

```diff
@@ -0,0 +1,159 @@
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.cluster.sharding.ShardCoordinator.LeastShardAllocationStrategy
+import akka.typed.{ ActorRef, ActorSystem, Props, TypedSpec }
+import akka.typed.cluster.Cluster
+import akka.typed.internal.adapter.ActorSystemAdapter
+import akka.typed.scaladsl.Actor
+import akka.typed.scaladsl.adapter._
+import akka.typed.testkit.TestKitSettings
+import akka.typed.testkit.scaladsl.TestProbe
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+
+object ClusterShardingSpec {
+  val config = ConfigFactory.parseString(
+    """
+      akka.actor.provider = cluster
+      
+      // akka.loglevel = debug
+      
+      akka.remote.artery.enabled = true
+      akka.remote.netty.tcp.port = 0
+      akka.remote.artery.canonical.port = 0
+      
+      akka.cluster.jmx.multi-mbeans-in-same-jvm = on
+      
+      akka.coordinated-shutdown.terminate-actor-system = off
+      
+      akka.actor {
+        serialize-messages = off
+        allow-java-serialization = off
+      }
+    """.stripMargin
+  )
+
+  sealed trait TestProtocol
+  final case class ReplyPlz(toMe: ActorRef[String]) extends TestProtocol
+  final case class WhoAreYou(replyTo: ActorRef[String]) extends TestProtocol
+  final case class StopPlz() extends TestProtocol
+
+  sealed trait IdTestProtocol { def id: String }
+  final case class IdReplyPlz(id: String, toMe: ActorRef[String]) extends IdTestProtocol
+  final case class IdWhoAreYou(id: String, replyTo: ActorRef[String]) extends IdTestProtocol
+  final case class IdStopPlz(id: String) extends IdTestProtocol
+
+}
+
+class ClusterShardingSpec extends TypedSpec(ClusterShardingSpec.config) with ScalaFutures {
+  import ClusterShardingSpec._
+
+  implicit val system = adaptedSystem
+  implicit val testkitSettings = TestKitSettings(system)
+  val sharding = ClusterSharding(adaptedSystem)
+
+  implicit val untypedSystem = ActorSystemAdapter.toUntyped(adaptedSystem)
+  private val untypedCluster = akka.cluster.Cluster(untypedSystem)
+
+  val behavior = Actor.immutable[TestProtocol] {
+    case (_, StopPlz()) ⇒
+      Actor.stopped
+
+    case (ctx, WhoAreYou(replyTo)) ⇒
+      replyTo ! s"I'm ${ctx.self.path.name}"
+      Actor.same
+
+    case (_, ReplyPlz(toMe)) ⇒
+      toMe ! "Hello!"
+      Actor.same
+  }
+  val behaviorWithId = Actor.immutable[IdTestProtocol] {
+    case (_, IdStopPlz(_)) ⇒
+      Actor.stopped
+
+    case (ctx, IdWhoAreYou(_, replyTo)) ⇒
+      replyTo ! s"I'm ${ctx.self.path.name}"
+      Actor.same
+
+    case (_, IdReplyPlz(_, toMe)) ⇒
+      toMe ! "Hello!"
+      Actor.same
+  }
+
+  object `Typed cluster sharding` {
+
+    untypedCluster.join(untypedCluster.selfAddress)
+
+    def `01 must send messsages via cluster sharding, using envelopes`(): Unit = {
+      val ref = sharding.spawn(
+        behavior,
+        Props.empty,
+        "example",
+        ClusterShardingSettings(adaptedSystem),
```

Comment:
why `adaptedSystem`? don't we have typed settings?

Reply:
that is the typed system, will rename it to system